pkg/model/filters/filters.go

```diff
@@ -72,16 +72,17 @@ func SplitForReportersMerge(q SingleQuery) (SingleQuery, SingleQuery) {
 		}
 	}
 	// The rationale here is that most traffic is duplicated from ingress and egress PoV, except cluster-external traffic.
-	// Merging is done by running a first query with FlowDirection=EGRESS and another with FlowDirection=INGRESS AND SrcOwnerName is empty,
+	// Ingress traffic will also contains pktDrop and DNS responses.
+	// Merging is done by running a first query with FlowDirection=INGRESS and another with FlowDirection=EGRESS AND DstOwnerName is empty,
 	// which stands for cluster-external.
-	// (Note that we use SrcOwnerName both as an optimization as it's a Loki index,
+	// (Note that we use DstOwnerName both as an optimization as it's a Loki index,
 	// and as convenience because looking for empty fields won't work if they aren't indexed)
 	q1 := SingleQuery{
-		NewMatch(fields.FlowDirection, `"`+constants.Egress+`"`),
+		NewMatch(fields.FlowDirection, `"`+constants.Ingress+`"`),
 	}
 	q2 := SingleQuery{
-		NewMatch(fields.FlowDirection, `"`+constants.Ingress+`"`),
-		NewMatch(fields.SrcOwnerName, `""`),
+		NewMatch(fields.FlowDirection, `"`+constants.Egress+`"`),
+		NewMatch(fields.DstOwnerName, `""`),
```

Comment:
why would reversing the logic works now, I think u saw that same issue either way ?

Reply:
DNS response codes are in `INGRESS` flows so I want these first and then fill the gap with the `EGRESS` ones.
pktDrop seems to be more consistent but I also saw differences

Reply:
On top of that, `Source` and `Destination` filters doesn't consider `FlowDirection` in the plugin.

If you download a file from a pod and set `Destination Namespace`, you don't see your flows with the current implementation.

With the inverted one it works:
![image](https://github.com/netobserv/network-observability-console-plugin/assets/91894519/3acb0349-8afd-4eb9-9744-e177e46a7206)

Reply:
I'm investigating both download / upload from pod cases since I have issue with old implementation and new one

Reply:
Here are my tests for consistency:
- Download a ubuntu.iso of 1000.73 MB and observe transferts between the IP and Node
- Upload the downloaded image and see the differences with the new IP

```
--- download ---
curl https://releases.ubuntu.com/22.04.2/ubuntu-22.04.2-desktop-amd64.iso --output /tmp/ubuntu.iso

reinterpret + Egress (old implementation)
pod 	1.2GB
node	1.3GB
=> same results using ingress

no reinterpret + Egress
pod 	922.7MB
node	625.1MB

no reinterpret + Ingress (new implementation)
pod 	9.2MB ???
node	630.4MB

--- upload ---
curl bashupload.com -T /tmp/ubuntu.iso

reinterpret + Egress (old implementation)
pod 	2.8GB
node	4.1GB
=> same results using ingress

no reinterpret + Egress
pod 	4.4MB ??
node	2.1GB

no reinterpret + Ingress (new implementation)
pod 	2.1GB
node	2.1GB
```

Note:
- If you `download` a file from a pod and set `Destination Namespace`, you don't see the expected bytes in topology view with my update. 
With the original one it worked:
![image](https://github.com/netobserv/network-observability-console-plugin/assets/91894519/3acb0349-8afd-4eb9-9744-e177e46a7206)

- If you `upload` a file from a pod and set `Source Namespace`, you see your flows with my update. 
With the original one it didn't worked:
![image](https://github.com/netobserv/network-observability-console-plugin/assets/91894519/5386ada9-78ca-4c26-965e-44d8729407df)

So in conclusion we can't do that simplification as is if we want to keep `Src` / `Dst` consistency and track both `download` & `upload` amounts !

Reply:
see https://github.com/netobserv/network-observability-console-plugin/pull/366#issuecomment-1686139968 solution :smile_cat: